### PR TITLE
Always build Debug and Release Poco

### LINKED ISSUE #11: support different build type
The "fat" binary archive generated for a release is build in release mode. When a user tries to build code on top in debug mode this package doesn't pick the built version of `Poco` but tries to find a debug version of the library on the system which might not be available.

Follow up of ros2/examples#199. @athackst FYI.

### LINKED ISSUE #199: No rule to make target '/usr/lib/libPocoFoundationd.so'
## Bug report

**Required Info:**

- Operating System:
  -  Ubuntu 16.04
- Installation type:
  -binaries
- Version or commit hash:
  - 5c635c95b3ae46314fe56d446c1c36293f5ccf2f
- DDS implementation:
  - default

#### Steps to reproduce issue
Steps I took on a clean install of Ubuntu 16.04
``` bash
$ sudo apt update && sudo apt install curl
$ curl http://repo.ros2.org/repos.key | sudo apt-key add -
$ sudo sh -c 'echo "deb [arch=amd64,arm64] http://repo.ros2.org/ubuntu/main xenial main" > /etc/apt/sources.list.d/ros2-latest.list'
$ sudo apt update
$ sudo apt install `apt list "ros-ardent-*" 2> /dev/null | grep "/" | awk -F/ '{print $1}' | grep -v -e ros-ardent-ros1-bridge -e ros-ardent-turtlebot2- | tr "\n" " "`
$ sudo apt install python-argcomplete 
$ source /opt/ros/ardent/share/ros2cli/environment/ros2-argcomplete.bash
$ source /opt/ros/ardent/setup.bash
$ mkdir -p ros2_ws/src
$ cd ros2_ws/src
$ git clone https://github.com/ros2/examples.git
$ ament build --cmake-args -DCMAKE_BUILD_TYPE=Debug
```

#### Expected behavior
successful build
#### Actual behavior
```bash
Scanning dependencies of target composition_nodes
make[2]: *** No rule to make target '/usr/lib/libPocoFoundationd.so', needed by 'libcomposition_nodes.so'.  Stop.
make[2]: *** Waiting for unfinished jobs....
[ 11%] Building CXX object CMakeFiles/composition_nodes.dir/src/publisher_node.cpp.o
[ 22%] Building CXX object CMakeFiles/composition_nodes.dir/src/subscriber_node.cpp.o
CMakeFiles/Makefile2:141: recipe for target 'CMakeFiles/composition_nodes.dir/all' failed
make[1]: *** [CMakeFiles/composition_nodes.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

<== Command '. /home/athackst/ros2_ws/build/examples_rclcpp_minimal_composition/cmake__build.sh && /usr/bin/make -j4 -l4' failed in '/home/athackst/ros2_ws/build/examples_rclcpp_minimal_composition' with exit code '2'
<== Command '. /home/athackst/ros2_ws/build/examples_rclcpp_minimal_composition/cmake__build.sh && /usr/bin/make -j4 -l4' failed in '/home/athackst/ros2_ws/build/examples_rclcpp_minimal_composition' with exit code '2'
```
#### Additional information
I was following the instructions here https://github.com/ros2/ros2/wiki/Ament-Tutorial minus some things that seemed related to a source build of ros2

I skipped the add some sources (https://github.com/ros2/ros2/wiki/Ament-Tutorial#add-some-sources), and went straight to the create an overlay (https://github.com/ros2/ros2/wiki/Ament-Tutorial#create-an-overlay)
If I'm doing a binary install do I still need to add those sources?
What did I miss?

### PR DESCRIPTION
**Pull request by sloretz**
This PR makes sure there is always a version of poco built with `CMAKE_BUILD_TYPE=Debug`. If `CMAKE_BUILD_TYPE` is `Debug` then it makes sure there is a version built with `Release`. Both get installed to the same place so there is always a library with and without a `d` postfix.

Tested by
1. Started from a clean install of ros2 ardent
1. Reproduced ros2/examples#199 (checked out examples branch `ardent` instead of master)
1. `sudo dpkg --force-depends --purge ros-ardent-poco-vendor`
1. Built this branch of poco_vendor in its own workspace and installed it to `/opt/ros/ardent`
    1. `ament build` (did not specify `CMAKE_BUILD_TYPE`)
    1. `sudo cp -R poco_ws/install/* /opt/ros/ardent/`
1. Clean build of examples workspace using `CMAKE_BUILD_TYPE=Debug` and confirmed it built successfully
1. Clean build of examples workspace using `CMAKE_BUILD_TYPE=Release` and confirmed it built successfully

Fixes #11 
Fixes ros2/examples#199

### CODE REVIEW

**Comment by sloretz** [comment]
CI

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4192/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1256/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3478/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4286/)
    * Failed with java traceback. Rebuild succeeded: [Build Status](http://ci.ros2.org/job/ci_windows/4288/)

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,11 +7,19 @@ set(PACKAGE_VERSION "1.0.0")
 list(INSERT CMAKE_MODULE_PATH 0 "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
 find_package(Poco COMPONENTS Foundation)
 
-if(NOT Poco_FOUND)
+
+function(get_poco)
+  set(options)
+  set(oneValueArgs BUILD_TYPE)
+  set(multiValueArgs)
+  cmake_parse_arguments(get_poco "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
```
Comment (wjwwood) [inline CMakeLists.txt]: nitpick: extra space before `)`

I'd remove it myself, but I didn't know if it was intentional or something.

**Review by wjwwood — APPROVED** [review]
lgtm, made one style commit and one nitpick comment I wasn't sure about.

File: CMakeLists.txt (hunk 1 above)
Comment (sloretz) [reply CMakeLists.txt]: Unintentional. Removed in 6ae93dcd7b5be88f2ec029e94ae19aaec35adcaf

**Comment by sloretz** [comment]
I retested locally with e1b621b and 6ae93dc and the issue still appears to be fixed. Merging without re-running CI since those two commits are just whitespace changes.

**Pull request merged**